src/pynwb/image.py

```diff
@@ -65,17 +71,25 @@ def __init__(self, **kwargs):
         if args_to_set['external_file'] is None and data is None:
             raise ValueError("Must supply either external_file or data to %s '%s'."
                              % (self.__class__.__name__, name))
-
         # data and unit are required in TimeSeries, but allowed to be None here, so handle this specially
         if data is None:
             kwargs['data'] = ImageSeries.DEFAULT_DATA
         if unit is None:
             kwargs['unit'] = ImageSeries.DEFAULT_UNIT
 
+        external_file_shape = get_data_shape(args_to_set['external_file'])
+        starting_frame_shape = get_data_shape(args_to_set['starting_frame'])
+        if external_file_shape is not None and starting_frame_shape is None:
+            if len(external_file_shape) == 1:
+                args_to_set['starting_frame'] = [0]
+        if external_file_shape != get_data_shape(args_to_set['starting_frame']):
+            warnings.warn(
+                "%s '%s': The number of frame indices in 'starting_frame' should have the same length "
+                "as 'external_file'." % (self.__class__.__name__, name)
+            )
```

Comment:
comments from @bendichter :
- this should be moved to a private function
- when `starting_frame` is `None`, we should have `[0] * len(args_to_set['external_file'])` (setting default not just for a signle external file)
- we should also have a check that ensures that `external_file` is only used when `format` is `"external"`

Reply:
> Setting starting_frame = [0] * len(args_to_set['external_file']) (i.e., setting it to [0, 0, 0, 0, ...]) is unfortunately not the correct value in almost all cases as it would mean that none of the external files contain any frames.

wait, what? @oruebel I think you and I have different ideas of what `starting_frame` means

Reply:
According to the schema, ``starting_frame`` is described as:

```
Each external image may contain one or more consecutive frames of the full ImageSeries. This attribute serves as an index to indicate which frames each file contains, to faciliate random access. The ‘starting_frame’ attribute, hence, contains a list of frame numbers within the full ImageSeries of the first frame of each file listed in the parent ‘external_file’ dataset. Zero-based indexing is used (hence, the first element will always be zero). For example, if the ‘external_file’ dataset has three paths to files and the first file has 5 frames, the second file has 10 frames, and the third file has 20 frames, then this attribute will have values [0, 5, 15]. If there is a single external file that holds all of the frames of the ImageSeries (and so there is a single element in the ‘external_file’ dataset), then this attribute should have value [0].
```

I.e., ``starting_frame`` describes for each ``external_file`` what index the first frame in that file corresponds to in the whole image series. In this way, if we want to access a particular frame,  ``starting_frame`` allows us to identify which ``external_file`` contains that frame and what the index of that frame within the external file is. Without this information one would need to open all external files to figure out how many frames each contains in order to find a particular frame.

Reply:
oh, I see. OK I guess I was mistaken about what `starting_frame` was doing.